cmd/oras/root/push.go

```diff
@@ -97,6 +101,9 @@ Example - Push repository with manifest annotations:
 Example - Push repository with manifest annotation file:
   oras push --annotation-file annotation.json localhost:5000/hello:v1
 
+Example - Push repository with platform:
+  oras push --platform linux/arm/v5 localhost:5000/hello:v1
```

Comment:
CC @FeynmanZhou for the flag naming. `attach` already using it so better call the new one `--config-platform`

Reply:
It is a good point that `--platform` means differently in `oras push` and `oras attach`. We need a better flag name.

Reply:
`--config-platform` does not sound like a good name. Would it be better to name it `--artifact-platform`? That will be consistent with `--artifact-type`.

Reply:
This does mean pull will use `--platform` and push `--artifact-platform` though which does seem a bit inconsistent to me.

Reply:
It's hard to choose a good flag name.

Some historical view on this: `--platform` was firstly introduced to `oras manifest to select an image with specific platform from a multi-arch image. Then the same UX was applied to `oras discover` and `oras attach` for selecting a platform-specific subject artifact.